.github/workflows/automated_release.yml

```diff
@@ -38,17 +46,25 @@ jobs:
         run: |
           bundle config path vendor/bundle
           bin/setup
+
       # Actually run our build
       - name: Compile Build
         run: ./compile-build
-      # https://github.com/marketplace/actions/upload-a-release-asset
-      - name: Upload release.gz
-        id: upload-release-gz
-        uses: actions/upload-release-asset@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # Create a release tag based on the branch name and .release-version file
+      - name: Set release tag
+        # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
+        # On the develop branch this might create RELEASE_VERSION=2.4.6-987654321-develop
+        # On the master branch this would then only create RELEASE_VERSION=2.4.6
+        run: echo "RELEASE_VERSION=$(printf -- '%s%s\n' $(cat .release-version) $([ ${BRANCH_NAME} = "develop" ] && printf -- '-%s-develop' ${GITHUB_RUN_ID} || echo ""))" >> $GITHUB_ENV
+
+      # Create a GitHub release with the release asset as an artifact
+      - name: Create release and upload release.tar.gz
+        uses: ncipollo/release-action@v1.8.8
         with:
-          upload_url: ${{ github.event.release.upload_url }} # Pull the URL from the event
-          asset_path: ./release.tar.gz
-          asset_name: release.tar.gz
-          asset_content_type: application/gzip
+          name: ${{ env.RELEASE_VERSION }}
+          tag: v${{ env.RELEASE_VERSION }}
```

Comment:
Or update the regex, means some slight historical inconsistency, but I guess things would then be consistent between repos.

Reply:
I think we should go with cross-repository consistency.  Not just for the consistency, but because GitHub are advocating for the use of semantic versioning in release tags.  They use this for sort order on the releases screen and it's good practice:

> It’s common practice to prefix your version names with the letter `v`. Some good tag names might be `v1.0` or `v2.3.4`.
> 
> If the tag isn’t meant for production use, add a pre-release version after the version name. Some good pre-release versions might be `v0.2-alpha` or `v5.9-beta.3`.

Reply:
As for the release notes, there are options.  For example:

- All releases are editable after they're created, so if you want release notes on it, you can add them any time.
- You could have a `.release-notes` file for builds coming from `master`.  The release action in our workflow supports adding release notes to the release and can read them from a file.
    - If you wanted to do that, you could make it be another check performed on push to `master`.  Reject pushes that don't update the release notes.

So I guess that's three options.

Reply:
Thing I'd prefer to look at auto-generating them from commits if we're going the way of automating things, otherwise the process will take even more steps then the previous.

Reply:
What was the previous process?  I don't know how hard it would be to generate release notes from commits, but I would say that it is a bit beyond what this task intended and may be something that should be looked into separately.  I also am not sure that commit messages should be so tightly coupled to release notes.  Often commits are going back and forth a bit and use technical jargon that end users don't want to know about.  They're also supposed to be written in a certain way and to a certain length that are both restrictions that don't apply to release notes.

Reply:
Just create the release from the releases tag which would generate the build, and fill in the release notes there.

When I say from the commits, I don't mean pulling in the raw commit messages, but either:
1) Identifying the stories associated those commits, and linking those
2) Using annotations in the commits to build the changelog. A common convention is using something like:

[FIX] Sequencescape now correctly does what you tell it to.

Wasn't suggesting it for now, just something to consider for the future.

Reply:
Ah I see 😄 yes those both sound like they need some careful thinking about if that should be automated.  Hopefully what we have in this PR is a step in the right direction.